reanahub/reana
# helm: optional max user-custom memory limits for k8s jobs

closes reanahub/reana-client#496

#### To test

- Comment out `{{- $_ := required "In order to use .Values.kubernetes_jobs_max_user_memory_limit safely you should set the label .Values.node_label_runtime and label a node with it." .Values.node_label_runtime }}` line if your using a single-node cluster.
- Add `kubernetes_jobs_max_user_memory_limit` to your `values.yaml`. If this value is lower than what the user set as limit per job via `reana.yaml`'s `kubernetes_memory_limit`, the job should fail and an explanatory message should be displayed as part of `reana-client logs`. E.g.

```diff
diff --git a/helm/reana/values.yaml b/helm/reana/values.yaml
index c5bfe9e..12fc5d1 100644
--- a/helm/reana/values.yaml
+++ b/helm/reana/values.yaml
@@ -114,3 +114,5 @@ traefik:
   ssl:
     enabled: true
     generateTLS: true
+
+kubernetes_jobs_max_user_memory_limit: 1G
```

```console
# with `kubernetes_memory_limit: 1Gi`
$ reana-client run
....
$ reana-client logs
==> Workflow engine logs
Workflow exited unexpectedly.
Job submission error: The "kubernetes_memory_limit" provided (1Gi) exceeds the limit (1G).
```
